docs/source/en/model_doc/superglue.md

```diff
@@ -0,0 +1,149 @@
+url = "https://github.com/magicleap/SuperGluePretrainedNetwork/blob/master/assets/phototourism_sample_images/london_bridge_19481797_2295892421.jpg?raw=true"
+im2 = Image.open(requests.get(url, stream=True).raw)
+images = [im1, im2]
+
+processor = AutoImageProcessor.from_pretrained("stevenbucaille/superglue_outdoor")
+model = AutoModel.from_pretrained("stevenbucaille/superglue_outdoor")
+
+inputs = processor(images, return_tensors="pt")
+outputs = model(**inputs)
+```
+
+The outputs contain the list of keypoints detected by the keypoint detector as well as the list of matches with their corresponding matching scores.
+Due to the nature of SuperGlue, to output a dynamic number of matches, you will need to use the mask attribute to retrieve the respective information:
+
+```python
+from transformers import AutoImageProcessor, AutoModel
+import torch
+from PIL import Image
+import requests
+
+url_image_1 = "https://github.com/cvg/LightGlue/blob/main/assets/sacre_coeur1.jpg?raw=true"
+image_1 = Image.open(requests.get(url_image_1, stream=True).raw)
+url_image_2 = "https://github.com/cvg/LightGlue/blob/main/assets/sacre_coeur2.jpg?raw=true"
+image_2 = Image.open(requests.get(url_image_2, stream=True).raw)
+
+images = [image_1, image_2]
+
+processor = AutoImageProcessor.from_pretrained("stevenbucaille/superglue_outdoor")
+model = AutoModel.from_pretrained("stevenbucaille/superglue_outdoor")
+
+inputs = processor(images, return_tensors="pt")
+with torch.no_grad():
+    outputs = model(**inputs)
+
+# Get the respective image masks 
+image0_mask, image1_mask = outputs_mask[0]
+
+image0_indices = torch.nonzero(image0_mask).squeeze()
+image1_indices = torch.nonzero(image1_mask).squeeze()
+
+image0_matches = outputs.matches[0, 0][image0_indices]
+image1_matches = outputs.matches[0, 1][image1_indices]
+
+image0_matching_scores = outputs.matching_scores[0, 0][image0_indices]
+image1_matching_scores = outputs.matching_scores[0, 1][image1_indices]
+```
+
+You can then print the matched keypoints on a side-by-side image to visualize the result :
+```python
+import cv2
+import numpy as np
+
+# Create side by side image
+input_data = inputs['pixel_values']
+height, width = input_data.shape[-2:]
+matched_image = np.zeros((height, width * 2, 3))
+matched_image[:, :width] = input_data.squeeze()[0].permute(1, 2, 0).cpu().numpy()
+matched_image[:, width:] = input_data.squeeze()[1].permute(1, 2, 0).cpu().numpy()
+matched_image = (matched_image * 255).astype(np.uint8)
+
+# Retrieve matches by looking at which keypoints in image0 actually matched with keypoints in image1
+image0_mask = outputs.mask[0, 0]
+image0_indices = torch.nonzero(image0_mask).squeeze()
+image0_matches_indices = torch.nonzero(outputs.matches[0, 0][image0_indices] != -1).squeeze()
+image0_keypoints = outputs.keypoints[0, 0][image0_matches_indices]
+image0_matches = outputs.matches[0, 0][image0_matches_indices]
+image0_matching_scores = outputs.matching_scores[0, 0][image0_matches_indices]
```

Comment:
This is redefining objects we have above e.g. image0_indices and image0_mask

Reply:
Removed unnecessary variables that are already defined in the first code block